src/Umbraco.Tests.Integration/Umbraco.Infrastructure/Services/KeyValueServiceTests.cs

```diff
@@ -19,6 +20,27 @@ public class KeyValueServiceTests : UmbracoIntegrationTest
     {
         private IKeyValueService KeyValueService => GetRequiredService<IKeyValueService>();
 
+        [Test]
+        public void Can_Query_For_Key_Prefix()
+        {
+            // Arrange
+            KeyValueService.SetValue("test1", "hello1");
+            KeyValueService.SetValue("test2", "hello2");
+            KeyValueService.SetValue("test3", "hello3");
+            KeyValueService.SetValue("test4", "hello4");
+
```

Comment:
Just to make it even more clear, that the service do not just return everything
```suggestion
            KeyValueService.SetValue("test4", "hello4");
            KeyValueService.SetValue("someotherprefix1", "helloagain1");
```